aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -709,21 +713,68 @@ private void launchSignOut(final String redirectUri, final String browserPackage
         launchCustomTabs(fqdn, null, browserPackage);
     }
 
+    /***
+     * Check if a browser is installed on the device to launch HostedUI.
+     * @return true if a browser exists else false.
+     */
+    public Boolean isBrowserInstalled() {
```

Comment:
This method should cache the result so that we aren't resolving this activity more than once.
Refer to similar logic being handled [here](https://github.com/aws-amplify/aws-sdk-android/blob/74209c4914f11a0476025e0d656e9182e56d1506/aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java#L775-L778).